modules/shared/src/main/scala/org/tessellation/schema/swap.scala

```diff
@@ -0,0 +1,72 @@
+package org.tessellation.schema
+
+import org.tessellation.schema.address.Address
+import org.tessellation.schema.transaction.{TransactionAmount, TransactionSalt}
+import org.tessellation.security.hash.Hash
+
+import derevo.cats.{order, show}
+import derevo.circe.magnolia.{decoder, encoder}
+import derevo.derive
+import eu.timepit.refined.types.numeric.PosLong
+import io.estatico.newtype.macros.newtype
+
+object swap {
+
+  @derive(decoder, encoder, order, show)
+  @newtype
+  case class AllowSpendFee(value: PosLong)
+
+  @derive(decoder, encoder, order, show)
+  @newtype
+  case class CurrencyId(value: Option[Address])
```

Comment:
How the `CurrencyId` may have an empty `value`? Isn't it non-existent `CurrencyId`?
IMO it should be an `Option[CurrencyId]` in the `AllowSpend` class if the currencyId is an optional type

Reply:
Empty should indicate DAG, either level of option representation should convey the same thing, either it's a Some(None) or None for DAG indicator